nestedtensor/nested/functions.py

```diff
@@ -141,29 +133,11 @@ def _embedding_bag(input, *args, **kwargs):
 
 orig_batch_norm = torch.nn.functional.batch_norm
 
-
 def batch_norm(input, running_mean, running_var, weight=None, bias=None, training=False, momentum=0.1, eps=1e-05):
-    if utils.find_nested_tensor_dispatch_key(input, running_mean, running_var, weight, bias, training, momentum, eps) is None:
-        return orig_batch_norm(input, running_mean, running_var, weight, bias, training, momentum, eps)
-
-    def t_batch_norm(input: torch.Tensor, running_mean: torch.Tensor, running_var: torch.Tensor, weight, bias, training, momentum, eps):
-        squeeze_after = False
-        # TODO: Need support for BatchNorm1d and BatchNorm2d as well
-        if input.dim() == 3:
-            # Check if is single image (leading batch dimension removed)
-            if input.size(1) != running_mean.size(0):
-                input = input.unsqueeze(0)
-                squeeze_after = True
-        result = orig_batch_norm(
-            input, running_mean, running_var, weight, bias, training, momentum, eps)
-        if squeeze_after:
-            result = result.squeeze(0)
-        return result
-
-    tw_batch_norm = utils.tensorwise()(t_batch_norm)
-
-    return tw_batch_norm(input, running_mean, running_var, weight, bias, training, momentum, eps)
-
+    validate_nt(input)
```

Comment:
You also want to make sure the other arguments are either Tensors or NestedTensors as appropiate